asllib/SideEffect.ml

```diff
@@ -0,0 +1,512 @@
+type identifier = AST.identifier
+
+module ISet = ASTUtils.ISet
+module IMap = ASTUtils.IMap
+
+module TimeFrame = struct
+  type t = Constant | Config | Execution of bool
+
+  let equal t1 t2 =
+    match (t1, t2) with
+    | Constant, Constant | Config, Config -> true
+    | Execution b1, Execution b2 -> Bool.equal b1 b2
+    | Constant, (Config | Execution _)
+    | Config, (Constant | Execution _)
+    | Execution _, (Config | Constant) ->
+        false
+
+  let is_before t1 t2 =
+    match (t1, t2) with
+    | Constant, Constant | Config, Config | Execution _, Execution _ -> true
```

Comment:
So both `Execution(true) < Execution(false)` and `Execution(false) < Execution(true)`, which means this is not a total order. It's a little strange.